lib/airbrake/rails/controller_methods.rb

```diff
@@ -57,7 +57,14 @@ def rails3?
 
       def filter_rails3_parameters(hash)
         ActionDispatch::Http::ParameterFilter.new(
-          ::Rails.application.config.filter_parameters).filter(hash)
+          ::Rails.application.config.filter_parameters
+        ).filter(recursive_stringify_keys(hash))
+      end
+
+      def recursive_stringify_keys(hash)
+        hash = hash.stringify_keys
+        hash.each {|k,v| hash[k] = recursive_stringify_keys(v) if v.is_a?(Hash) && v.respond_to?(:stringify_keys) } # Rack::Session::Abstract::SessionHash has a stringify_keys method we should not call
```

Comment:
This line could do with a bit of decompression. Would it be possible for you to make it a bit more readable, adding some newlines at least?